lib/pkcs11_req.c

```diff
@@ -1133,6 +1146,8 @@ int pkcs11_sign_X509_REQ(pkcs11Context * p11Context, CK_VOID_PTR req, int output
     {
 
 	DSA_SIG *sig = DSA_SIG_new();
+	BIGNUM *sig_r;
```

Comment:
Since `sig_r`and `sig_s` are used for both `CKM_ECDSA_XXXX` and `CKM_DSA_SHA1`, and taking into account memory cleanup changes (see other comments), these pointers must be declared in the prolog of the function (around line 844) and assigned to `NULL`.